[cmake] Add cmake parameter to BuildSetup.bat

fetzerch opened the pull request:
First attempt to integrate cmake into the BuildSetup.bat script. This is not perfect as the packaging only takes the main executable from the cmake build and nothing more. Still good enough for the beginning to verify the build on Jenkins.

Would be happy to get some reviews from people that know batch scripting.
@paxxi, @Montellese, @ace20022 maybe

fetzerch committed:
[cmake] Add cmake parameter to BuildSetup.bat

First attempt to integrate cmake into the BuildSetup.bat script.
This is not perfect as the packaging only takes the main executable from
the cmake build and nothing more. Still good enough to verify on Jenkins.

MartijnKaijser commented:
Needs updating now we have Vs2015?

fetzerch commented:
Yep, seen it too late. Done.

Paxxi commented on project/Win32BuildSetup/BuildSetup.bat:
```diff
@@ -157,6 +161,41 @@ set WORKSPACE=%CD%\..\..
+  MKDIR %WORKSPACE%\kodi-build
+  PUSHD %WORKSPACE%\kodi-build
+
+  call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" x86
```
don't think this is needed, cmake takes care of finding the build tools itself. If it must be here, should probably use %VS140COMNTOOLS%\..\..\VC\vcvarsall.bat x86

fetzerch replied:
True, it's not needed, works well with the `-G "Visual Studio 14"` parameter alone.

fetzerch committed:
[cmake] Print source and binary dir paths

fetzerch committed:
[cmake] Fix compilation with msbuild (vs2015)

The CMake VS2015 generator sets the /Fd parameter (program database file
name) to <targetname>.pdb while the VS2014 generator used vc140.pdb.
Since we use precompiled headers over multiple static libraries we have to
use one single pdb file.

This fixes the errors:
fatal error C1083: Cannot open precompiled header file:
'C:/jenkins/slave/workspace/WIN-32/kodi-build/Release/objs/kodi.pch': No
such file or directory

See https://github.com/fetzerch/xbmc/commit/f626edf87a5af6c1f10d13d2d3b8103d6c2aadd1
for more information.

fetzerch committed:
[cmake] Switch back to Effects11_2013.sln

The 2013 version of the Effects11 project is still used after the switch
to VS2015. Use this version also with cmake to improve compatibility.

The Effects11 2015 solution has for example different output paths
configured and possibly more.

fetzerch committed:
[cmake] Adapt README.md for VS2015

fetzerch commented:
Successful build http://jenkins.kodi.tv/job/WIN-32/8481/ (with unrelated addon failure).
